source/Cosmos.System2/FileSystem/CosmosVFS.cs

```diff
@@ -821,7 +737,11 @@ public override bool IsValidDriveId(string driveId)
         /// <exception cref="Exception">Unable to determine filesystem for path:  + aDriveId</exception>
         public override long GetTotalSize(string aDriveId)
         {
-            var xFs = GetFileSystemFromPath(aDriveId);
+            var xFs = GetPartitionFromPath(aDriveId).MountedFS;
```

Comment:
Can you please refractor the getting and checking of the filesystem into a helper function to reduce repeated code?